### PR TITLE
removed duplicated training samples, fixed fold calculation

### LINKED ISSUE #288: Training on demo-rasa.json results in UndefinedMetricWarning
**rasa NLU version** (e.g. `0.7.3`):

**Used backend / pipeline** (`spacy_sklearn`):

**Operating system** (Ubuntu 64bit ):

**demo-rasa.json has duplicate training data*:
https://github.com/golastmile/rasa_nlu/blob/master/data/examples/rasa/demo-rasa.json

I just noticed that demo-rasa.json has duplicate data in "common_examples". I see that the whole examples are copied exactly twice.
I looked at the history of the file and found that this happened when the two sections "entity_examples" and "intent_examples" were merged into "common_examples" recently (btw both sections had the same data)

But now if I remove the duplicates from "common_examples" and train the model, I get the following warnings

`....../sklearn/metrics/classification.py:1113: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no predicted samples.
  'precision', 'predicted', average, warn_for)
....../sklearn/metrics/classification.py:1115: UndefinedMetricWarning: F-score is ill-defined and being set to 0.0 in labels with no true samples.
  'recall', 'true', average, warn_for)`

Of course the confidence level in results also get affected badly, e.g. trying with "chinese restaurant in the center" doesn't classify "cuisine" entity and the confidence level drops to under 4

If I keep the duplicate data it runs fine and without any warnings... This behavior is strange to me, can someone please explain why the data duplication is required?

Thanks

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- removed duplicates from rasa example training data
- added some more examples
- changed calculation of sklearn folds (won't matter for training sets with >25 examples per intent, for smaller datasets we will now choose fewer folds to avoid sklearn warnings)
- fixes #288 

**Status**:
- [x] ready for code review
- [x] there are tests for the functionality
- [x] documentation updated
- [x] changelog updated

### CODE REVIEW

File: rasa_nlu/classifiers/sklearn_intent_classifier.py [hunk 1]
```diff
@@ -81,7 +84,7 @@ def train(self, training_data, intent_features, num_threads):
 
         # dirty str fix because sklearn is expecting str not instance of basestr...
         tuned_parameters = [{'C': [1, 2, 5, 10, 20, 100], 'kernel': [str('linear')]}]
-        cv_splits = max(2, min(5, np.min(np.bincount(y))))
+        cv_splits = max(2, min(MAX_CV_FOLDS, np.min(np.bincount(y)) / 5))  # aim for at least 5 examples in each fold
```
Comment (twerkmeister) [inline rasa_nlu/classifiers/sklearn_intent_classifier.py]: X / Y is not truncated division anymore since we are using python 3 semantic, so between 2 and 5 we could end up with float numbers

Comment (tmbo) [inline rasa_nlu/classifiers/sklearn_intent_classifier.py]: right :+1:, fixed
